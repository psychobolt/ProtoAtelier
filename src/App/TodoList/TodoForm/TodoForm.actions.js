let nextTodoId = 0;

export const Actions = {
  ADD_TODO: 'addTodo',
};

export const addTodo = text => ({
  type: Actions.ADD_TODO,
  todo: {
    id: nextTodoId++, // eslint-disable-line no-plusplus
    text,
  },
});
