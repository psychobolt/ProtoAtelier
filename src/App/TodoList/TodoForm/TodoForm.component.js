// @flow
import React from 'react';
// @flow
import { XInput, type XInputEvent } from 'Framework/ReactXelToolkit';
import styles from './TodoForm.style';

type Props = {
  inputValue: string,
  onValueSubmit: (value: string) => void,
  style: {},
};

type State = {
  inputValue: string,
}

export const KEYCODE_ENTER = 13;

export default class TodoForm extends React.Component<Props, State> {
  static defaultProps = {
    inputValue: '',
    style: {},
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      inputValue: props.inputValue,
    };
  }

  onButtonClick = () => {
    if (this.state.inputValue) {
      this.submit();
    }
  }

  onInputKeyup = (event: XInputEvent) => {
    if (event.keyCode === KEYCODE_ENTER) {
      this.submit();
    } else {
      this.setState({
        inputValue: event.target.value,
      });
    }
  }

  submit() {
    this.props.onValueSubmit(this.state.inputValue);
    this.setState({
      inputValue: '',
    });
  }

  props: Props;

  render() {
    return (
      <x-box style={this.props.style}>
        <XInput
          style={styles.input}
          value={this.state.inputValue}
          onKeyup={this.onInputKeyup}
        />
        <x-button onClick={this.onButtonClick}>
          <x-label>Add Todo</x-label>
        </x-button>
      </x-box>
    );
  }
}
