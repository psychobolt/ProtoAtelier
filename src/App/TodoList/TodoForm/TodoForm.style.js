export default {
  input: {
    marginRight: '6px',
  },
};
