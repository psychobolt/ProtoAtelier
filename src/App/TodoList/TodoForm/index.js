export * from './TodoForm.actions';
export { default } from './TodoForm.container';
