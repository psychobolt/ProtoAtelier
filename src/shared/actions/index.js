export const Actions = {
  UNDO: 'undo',
  REDO: 'redo',
};

export const undo = () => ({
  type: Actions.UNDO,
});

export const redo = () => ({
  type: Actions.REDO,
});
