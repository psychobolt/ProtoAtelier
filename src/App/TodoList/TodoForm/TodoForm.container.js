// @flow
import { connect } from 'react-redux';
import type { Dispatch } from 'redux';

import TodoForm from './TodoForm.component';
import { addTodo } from './TodoForm.actions';

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  onValueSubmit: (value) => {
    if (value.trim()) {
      dispatch(addTodo(value));
    }
  },
});

export default connect(undefined, mapDispatchToProps)(TodoForm);
