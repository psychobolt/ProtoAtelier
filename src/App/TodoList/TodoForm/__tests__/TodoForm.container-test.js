import React from 'react';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import PropTypes from 'prop-types';

import TodoForm from '../TodoForm.container';
import { Actions } from '../TodoForm.actions';

const mockStore = configureMockStore([]);

const todos = { present: [] };

describe('container <TodoForm />', () => {
  it('should render without crashing', () => {
    shallow(<TodoForm />, {
      context: { store: mockStore({}) },
    });
  });

  it('should not add Todo if input value is undefined', () => {
    const store = mockStore({ todos });
    const wrapper = mount(<TodoForm />, {
      context: { store },
      childContextTypes: { store: PropTypes.object }, // eslint-disable-line react/forbid-prop-types
    });
    wrapper.find('x-button').simulate('click');
    expect(store.getState()).toEqual({ todos });
  });

  it('should not add Todo if input value is a empty string', () => {
    const store = mockStore({ todos });
    const wrapper = mount(<TodoForm inputValue=" " />, {
      context: { store },
      childContextTypes: { store: PropTypes.object }, // eslint-disable-line react/forbid-prop-types
    });
    wrapper.find('x-button').simulate('click');
    expect(store.getState()).toEqual({ todos });
  });

  it('should add Todo if input value is not empty', () => {
    const store = mockStore({ todos });
    const text = 'Item';
    const wrapper = mount(<TodoForm inputValue={text} />, {
      context: { store },
      childContextTypes: { store: PropTypes.object }, // eslint-disable-line react/forbid-prop-types
    });
    wrapper.find('x-button').simulate('click');
    expect(store.getActions()).toEqual([{
      type: Actions.ADD_TODO,
      payload: {
        id: 0,
        text,
      },
    }]);
  });
});
