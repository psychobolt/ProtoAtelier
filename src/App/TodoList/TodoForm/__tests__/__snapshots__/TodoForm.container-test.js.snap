// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container <TodoForm /> should render correctly 1`] = `
<TodoForm
  inputValue=""
  onValueSubmit={[Function]}
  style={Object {}}
/>
`;
